### PR TITLE
PERF-5120 Add two multiplanner focused clustered collection workloads

### PR DESCRIPTION
**Pull request by Fefer-Ivan**
**Jira Ticket:** PERF-5120

**Whats Changed:**  
ClusteredCollection workload that has huge _id.
UseClusteredIndex workload that has predicate on _id as selective as predicate on x1, making clustered index tied with x1 index.

**Patch testing results:**  
https://spruce.mongodb.com/version/65fc278f3a0f5b000781f94d/tasks

Note: it depends on the monotonic loader PR #1155

### CODE REVIEW

File: src/workloads/query/multiplanner/ClusteredCollection.yml [hunk 1]
```diff
@@ -0,0 +1,410 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
```
Comment (chiihuang) [inline src/workloads/query/multiplanner/ClusteredCollection.yml:3]: Update the test description to reflect how it tests with cluster index.

File: src/workloads/query/multiplanner/UseClusteredIndex.yml [hunk 2]
```diff
@@ -0,0 +1,410 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
```
Comment (chiihuang) [inline src/workloads/query/multiplanner/UseClusteredIndex.yml:3]: ditto

File: src/workloads/query/multiplanner/UseClusteredIndex.yml [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+    Key: InsertTemplate
+    Parameters:
+      database: *db
+      threads: 1
+      active: [1]
+      nopInPhasesUpTo: *maxPhase
+      repeat: 1
+      collection: *coll
+      docCount: *docCount
+      document: {
```
Comment (chiihuang) [inline src/workloads/query/multiplanner/UseClusteredIndex.yml:73]: [question] Does this also needs `_id: {^FastRandomString: {length: 512}},` for large field ? Or is it intended to make `_id` and `x` share similar size?

File: src/workloads/query/multiplanner/ClusteredCollection.yml [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+  Threads: 1
+  Database: *db
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *maxPhase
+      PhaseConfig:
+        Repeat: 1
+
+- Name: MultiplannerWith64Indexes
```
Comment (chiihuang) [inline src/workloads/query/multiplanner/ClusteredCollection.yml:278]: Make the actor name unique among tests such as `MultiplannerWith64IndexesClusteredCollection`. Same for the 16 and 2 indexes ones in this workload file.

**Review by chiihuang — CHANGES_REQUESTED** [review]
Overall looks good. Just a few change requests on comments and a question regarding the large field in UseClusteredIndex workload.

File: src/workloads/query/multiplanner/UseClusteredIndex.yml (hunk 3 above)
Comment (Fefer-Ivan) [reply src/workloads/query/multiplanner/UseClusteredIndex.yml:73]: In this workload we want to use clustered index, so incremental _id's, generated by the monotonic loader should be good.

File: src/workloads/query/multiplanner/ClusteredCollection.yml (hunk 4 above)
Comment (Fefer-Ivan) [reply src/workloads/query/multiplanner/ClusteredCollection.yml:278]: I am not sure if we want to do that.
In performance data the name of the workload would also be present, so in the end we will get something like
UserClusteredIndex.MultiplannerWith64Indexes - which is pretty clear. 
Adding a second mention of clustered index sounds like an overkill

Comment (dhsegel) [reply src/workloads/query/multiplanner/ClusteredCollection.yml:278]: I think either would work, but it should be consistent across all of our workloads. I have been giving each actor a distinct name up to now, but if we maintained consistency, the data might look something like this:
| variant                                              | task                          | test                           |
|------------------------------------------------------|-------------------------------|--------------------------------|
| perf-standalone-all-feature-flags.arm.aws.2023-11    | multiplanner_simple           | MultiplannerWith64Indexes.Crud |
| perf-standalone-classic-query-engine.arm.aws.2023-11 | multiplanner_simple           | MultiplannerWith64Indexes.Crud |
| perf-standalone-all-feature-flags.arm.aws.2023-11    | multiplanner_compound_indexes | MultiplannerWith64Indexes.Crud |
| perf-standalone-classic-query-engine.arm.aws.2023-11 | multiplanner_compound_indexes | MultiplannerWith64Indexes.Crud |

It effectively comes down to a question of whether we want to be referencing both `task` and `test` or whether we want to get all the identification information from `test`. I would see a benefit in maintaining both as we could filter down to workloads' data for `MultiplannerWith64Indexes` if that is something we want to do.

**Review by dhsegel — APPROVED** [review]
LGTM, barring description and naming updates.

File: src/workloads/query/multiplanner/ClusteredCollection.yml (hunk 4 above)
Comment (chiihuang) [reply src/workloads/query/multiplanner/ClusteredCollection.yml:278]: Makes sense. Either way seems good to me. @dpercy in terms of notebook analysis, which way is preferable? 

One benefit of avoiding second mentioning in data analysis is to group the tests across tasks, or having a table like:

| task \ test |  MultiplannerWith2Indexes | MultiplannerWith16Indexes | MultiplannerWith64Indexes |
| -- | -- | -- | -- |
| multiplanner_simple | 50ms | 51ms | 52ms |
| multiplanner_compound_indexes | 150ms | 200ms | 250ms |

Comment (Fefer-Ivan) [reply src/workloads/query/multiplanner/ClusteredCollection.yml:278]: I will follow the current pattern. Let's discuss during stand-up if we should do an update on that

Comment (Fefer-Ivan) [reply src/workloads/query/multiplanner/ClusteredCollection.yml]: Done

File: src/workloads/query/multiplanner/UseClusteredIndex.yml (hunk 2 above)
Comment (Fefer-Ivan) [reply src/workloads/query/multiplanner/UseClusteredIndex.yml:3]: Done

File: src/workloads/query/multiplanner/ClusteredCollection.yml (hunk 1 above)
Comment (Fefer-Ivan) [reply src/workloads/query/multiplanner/ClusteredCollection.yml:3]: Done

**Review by chiihuang — APPROVED** [review]

**Comment by Fefer-Ivan** [comment]
evergreen merge

**Pull request merged**